fitlins/base.py

```diff
@@ -145,23 +146,21 @@ def second_level(analysis, block, space, deriv_dir):
             job_desc.setdefault('contrasts', []).append(desc)
 
             if op.exists(stat_fname):
+                print(stat_fname)
                 continue
 
-            cols = {'intercept': np.ones(len(data))}
-            cname = 'intercept'
-            if not np.allclose(contrasts[contrast], 1):
-                cname = contrast
-                cols[contrast] = contrasts[contrast]
-
-            paradigm = pd.DataFrame(cols)
+            # Each contrast is computed as the intercept (omitting 0s)
+            intercept = contrasts[contrast]
+            i_data = list(np.array(data)[intercept != 0.0])
+            intercept = intercept[intercept != 0.0]
 
-            fmri_glm.fit(data, design_matrix=paradigm)
+            fmri_glm.fit(i_data,
+                         design_matrix=pd.DataFrame({'intercept': intercept}))
```

Comment:
No. We will have to revisit this later (maybe at the Hackathon), as it's a bit confusing how nistats works. But see this note in `compute_contrast`: Basically, at the second level its rare to do anything beside a simple one sample t-test. That covers 90+% of use cases.

            VERY IMPORTANT: The 'intercept' corresponds to the second level
            effect after taking confounders in consideration. If there are
            no confounders then this will be equivalent to a simple t test.
            By default we compute the 'intercept' second level contrast.

Reply:
Don't think of this as an intercept in the standard sense; it would be better if the argument were named `weights`. You can pass any pattern of weights in as the intercept, and it only acts like a traditional intercept in the special case where we want a one-sample t-test on a single set of maps (i.e., the "identity contrast" case).

Reply:
We just ran through this today, so I think it merits more investigation to see if this is the right way to do it. But at least for 1 vs 0 contrasts, it should be OK (which is again, most use cases, since people would compute complex contrasts (e.g. -1, 1), at the first level, then just want a one sample t-test of that contrast across the group)

Reply:
If you're asking whether we would ever want to pass additional design matrix columns in besides the weights, the answer is no--at least, not when computing contrasts. We *will* eventually have to implement handling of second-level models, and that's where we'll potentially have a full design matrix that includes multiple columns (including, e.g., second-level covariates like average performance for each subject).

Reply:
Yep, that's my understanding (that wasn't a question at you, just elaboration before I saw your comment)